### PR TITLE
Refactor orbital into class #13

### PR DESCRIPTION
**Pull request by johngian**

### CODE REVIEW

File: satnogsclient/observer/orbital.py [hunk 1]
```diff
@@ -1,56 +1,106 @@
...
+        # satellite object
+        if satellite_dict:
+            ret = self._set_satellite_dict(satellite_dict)
+            if not ret:
+                return False
+
+        # no problems
+        return True
+
+    def _set_observer(self, observer_dict):
```
Comment (johngian) [inline satnogsclient/observer/orbital.py:30]: Please use `getter/setter` paradigm like we do in the rest of the client.

File: satnogsclient/observer/orbital.py [hunk 2]
```diff
@@ -1,56 +1,106 @@
...
+        if satellite_dict:
+            if all([x in satellite_dict for x in ['tle0', 'tle1', 'tle2']]):
+                tle0 = str(satellite_dict['tle0'])
+                tle1 = str(satellite_dict['tle1'])
+                tle2 = str(satellite_dict['tle2'])
+                try:
+                    self.satellite = ephem.readtle(tle0, tle1, tle2)
+                    return True
+                except ValueError:
+                    #return {'ok': False,, 'error': 'ephem object - tle values problematic: {}'.format(sys.exc_info()[0])}
```
Comment (johngian) [inline satnogsclient/observer/orbital.py:56]: Please remove unused code in comments.

File: satnogsclient/observer/orbital.py [hunk 3]
```diff
@@ -1,56 +1,106 @@
...
+                    return False
+                except:
+                    #return {'ok': False,, 'error': 'ephem object: {}'.format(sys.exc_info()[0])}
```
Comment (johngian) [inline satnogsclient/observer/orbital.py:59]: Same here.

Comment (johngian) [inline satnogsclient/observer/orbital.py:59]: It doesn't really make sense to catch 2 different exceptions when we handle both the same way (eg. `return False`).

File: satnogsclient/observer/orbital.py [hunk 4]
```diff
@@ -1,56 +1,106 @@
...
+                    return False
+        return False
+
+    def pinpoint(self, observer_dict=None, satellite_dict=None, timestamp=None):
```
Comment (johngian) [inline satnogsclient/observer/orbital.py:63]: Is there any use case when we reassign `observer_dict` and `satellite_dict` on runtime? If not (or is corner case) let's just initialize dicts on `__init__` and allow changes via `setter/getter`.

**Comment by johngian** [comment]
This PR breaks the functionality here: https://github.com/satnogs/satnogs-client/blob/master/satnogsclient/observer/worker.py#L103
- You are not return the correct value when `not p[ok]`.
- You are not using the new class implementation.

**Comment by johngian** [comment]
Hey @jinxcat 
It looks like there are plenty of changes to be made. Either close this and reopen it when it is ready or provide an ETA for that.

File: satnogsclient/observer/orbital.py (hunk 3 above)
Comment (jinxcat) [inline satnogsclient/observer/orbital.py:59]: That's the thing... they _are_ different cases. If the return values are boolean, there is not enough flexibility to handle them differently on the calling function.

That's why I left the code in comments, returning dictionaries

File: satnogsclient/observer/orbital.py (hunk 4 above)
Comment (jinxcat) [inline satnogsclient/observer/orbital.py:63]: observer_dict can be reassigned in the case of a moving observer.

The satellite_dict could be reassigned if the observation is long enough that we have an updated TLE in the meantime. I think they do not add too much clutter, and they pave the way for using it to follow balloons

File: satnogsclient/observer/orbital.py (hunk 3 above)
Comment (johngian) [inline satnogsclient/observer/orbital.py:59]: Buy returning the same thing when catching a superset of the previous exception you are practically doing the (more complicated) equivalent of 

```
try:
   something
except:
   return False
```

File: satnogsclient/observer/orbital.py (hunk 4 above)
Comment (johngian) [inline satnogsclient/observer/orbital.py:63]: Sounds good. Having a moving observer is a corner case in our paradigm but I don't have any strong opinions on that.

**Comment by jinxcat** [comment]
> You are not return the correct value when not p[ok].

I am. All returns include 'ok', which is {True|False}. If False, it loops again.

> You are not using the new class implementation.

True, will fix. Was leaving this for 'observer_refactoring' branch, but needs to be in this PR

**Comment by johngian** [comment]
@jinxcat Thanks for the update. Let me know when the rest of the comments are outdated to give another look and merge.

**Comment by ppapadeas** [comment]
Closing this to clean up branches. Issue remains in backlog

**Pull request closed without merging**